modules/fedimint-ln-client/src/lib.rs

```diff
@@ -826,76 +825,6 @@ impl LightningClientModule {
             .map(|g| g.info)
     }
 
-    pub async fn select_active_gateway_opt(&self) -> Option<LightningGateway> {
-        match self.select_active_gateway().await {
-            Ok(gw) => Some(gw),
-            Err(e) => {
-                warn!(?e, "Could not select a gateway");
-                None
-            }
-        }
-    }
-
-    /// The set active gateway, or a random one if none has been set
-    pub async fn select_active_gateway(&self) -> anyhow::Result<LightningGateway> {
-        let mut dbtx = self.client_ctx.module_db().begin_transaction().await;
-        match dbtx.get_value(&LightningGatewayKey).await {
-            Some(active_gateway) => Ok(active_gateway.info),
-            None => {
-                let gateways = self.fetch_registered_gateways().await?;
-                let num_gateways = gateways.len();
-
-                let vetted = gateways
-                    .clone()
-                    .into_iter()
-                    .filter(|g| g.vetted)
-                    .collect::<Vec<_>>();
-                if !vetted.is_empty() {
```

Comment:
Another question that arises is how we want to handle the "vetted" field. Previously we were filtering non-vetted gateways when selecting the active gateway. Perhaps we should just leave it up to the integrating client.

Reply:
I think vetted gateways should be prioritized if provided and online, if not we should try another one. That would mean the "ping the GW before making a payment" logic would move into the selection function so we can skip offline vetted gateways.

This might lead to the "gateway cache" becoming its own service in the background that autonomously fetches available gateways and tries to connect to them to figure out which one to use when asked

Reply:
I can take this another direction if you like, but I opted to keep the API at the `fedimint-client` level as simple as possible. Lightning operations always expect a gateway to be passed in and callers can acquire this gateway either using `select_gateway` or implementing their own logic in combination with `list_gateways` (they can select vetted gateways this way).

The goal of this PR was to remove non-determinism from our tests and I think implementing a fallback gateway causes some issues (it was the source of https://github.com/fedimint/fedimint/issues/4350#issue-2139759549), since the client might unexpectedly choose a different gateway.

I think higher level apps can implement this fallback or select vetted gateways if they want, but the way it is now allows our tests to be more straightforward